types/vote.go

```diff
@@ -162,9 +165,9 @@ func (vote *Vote) ValidateBasic() error {
 			len(vote.ValidatorAddress),
 		)
 	}
-	if len(vote.Signature) == 0 {
-		return errors.New("signature is missing")
-	}
+	// if len(vote.Signature) == 0 {
+	// 	return errors.New("signature is missing")
+	// }
```

Comment:
The reason this is needed is that with this pr in order to create a signature we call fromproto which calls validate basic. This vote has a empty signature, therefore, causing an error. 

There are two options that I can see. We either do this (allow empty sigs) **or** make `vote.SignBytes` a function like `func SignVoteBytes(chainID string, vote *tmproto.Vote) []byte`, this will expect a vote that has a empty signature but when we call fromproto the signature will be filled.  The latter case causes quite a few more changes but does make it nicer since we are still expecting signatures not to be empty when checking. 

which is preferred?
cc @melekes @erikgrinaker

Reply:
Should `FromProto` call `ValidateBasic` then? It sounds like we need to vary the validation policy in different cases.

Of your listed options, I think I prefer the latter - but I don't fully understand the implications.

Reply:
I do think it should call ValidateBasic as its an safe measure before allowing the type to be worked on . The changes would need to be made everywhere we call SignBytes.

```
bz := vote.Signbytes(chainID)
```

would become

```
v := vote.Toproto()
bz := types.SignVoteBytes(chainID, v)
```